packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -330,6 +330,8 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
   ///
   /// The state of hovering pointers is not tracked because that would require
   /// hit-testing on every frame.
+  @protected
+  Map<int, HitTestResult> get hitTests => _hitTests;
```

Comment:
I don't think we should make this part of the public API.

Reply:
Hmm what about the "alternative soluution" described above?

> An alternative solution, which the flutter-smooth branch is currently using (but more hacky), may be to add a filter to dispatchEvent. Then, we can utilize the filter to skip those RenderObjects in the main tree.

Reply:
Or, as for this one: Is it acceptable to expose a *readonly* version? For example:

```dart
UnmodifiableMapView<int, UnmodifiableHitTestResultView> get hitTests => ...;
```

where unmodifiable map is https://api.flutter.dev/flutter/dart-collection/UnmodifiableMapView-class.html
and UnmodifiableHitTestResultView can be something like:

```dart
class UnmodifiableHitTestResultView {
  final HitTestResult _actual;
  ... only provide a few readonly method, so users have no possibility to write anything to it...
}
```